core/src/main/java/hudson/FilePath.java

```diff
@@ -2957,11 +2960,12 @@ private File deleting(File f) {
         return f;
     }
 
-    private boolean mkdirs(File dir) {
+    private boolean mkdirs(File dir) throws IOException {
         if (dir.exists())   return false;
```

Comment:
I feel this method should check whether the file is a directory and throw the exception otherwise. It is not related to this change, I will follow-up in a pull request. Maybe we can kill the method at all to be on the safe side